src/Modules/CalcSetup.lua

```diff
@@ -1052,7 +1059,14 @@ function calcs.initEnv(build, mode, override, specEnv)
 									gemData = gemInstance.gemData,
 								}
 								if gemInstance.gemData then
-									for _, value in ipairs(propertyModList) do
+									local socketColor = env.player.itemList[groupCfg.slotName] and env.player.itemList[groupCfg.slotName].sockets and env.player.itemList[groupCfg.slotName].sockets[gemIndex] and env.player.itemList[groupCfg.slotName].sockets[gemIndex].color
+									local socketedInColorCfg = nil
+									if socketColor then
+										socketedInColorCfg = {}
+										for k,v in pairs(groupCfg) do socketedInColorCfg[k] = v end
+										socketedInColorCfg.socketColor = socketColor
+									end
+									for _, value in ipairs(socketedInColorCfg and env.modDB:List(socketedInColorCfg, "GemProperty") or propertyModList) do
```

Comment:
Please keep consistency instead of using both `_, value` and `k,v`

Reply:
Not sure i understand. Do you want me to change `k,v` to `key,value` ?

Reply:
The better option is to just use `socketedInColorCfg = copyTable(groupCfg)`.  That's what it looks like you're doing here, anyway from just this snippet.  Probably doesn't work if you're not making socketedInColorCfg a new table first

Reply:
Didn't know that copyTable can make a shallow copy. i'll replace it with `copyTable(groupCfg, true)`.